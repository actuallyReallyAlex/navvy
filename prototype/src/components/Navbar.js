import React from 'react'
import { Box } from 'grommet'
import NavbarExtender from './NavbarExtender'
import NavbarLogo from './NavbarLogo'
import NavbarSearch from './NavbarSearch'
import NavbarNotifications from './NavbarNotifications'
import NavbarUserMenu from './NavbarUserMenu'

const Navbar = () => {
  return (
    <Box
      align="center"
      className="navbar"
      direction="row"
      elevation="small"
      height="70px"
      justify="between"
      pad="small"
    >
      <NavbarExtender />
      <NavbarLogo />
      <Box
        align="center"
        direction="row"
        justify="end"
        pad={{ right: 'medium' }}
        width="33%"
      >
        <NavbarSearch />
        <NavbarNotifications />
        <NavbarUserMenu />
      </Box>
    </Box>
  )
}

export default Navbar
