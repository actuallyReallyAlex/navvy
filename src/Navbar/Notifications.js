import React from 'react'
import { Box, Menu, Text } from 'grommet'
import { MailOption, Notification, Server } from 'grommet-icons'

const NavbarNotifications = () => {
  const notificationIsHovered = false; // ! Fix this

  return (
    <Box margin={{ horizontal: 'small' }}>
      <Menu
        dropAlign={{ top: 'bottom', right: 'right' }}
        icon={
          <Notification color={notificationIsHovered ? 'dark-2' : 'dark-6'} />
        }
        items={[
          {
            label: (
              <Box direction="row" gap="small" pad="small">
                <MailOption />
                <Text>You have 6 new messages</Text>
              </Box>
            ),
            onClick: () => {}
          },
          {
            label: (
              <Box direction="row" gap="small" pad="small">
                <Server />
                <Text>Server rebooted</Text>
              </Box>
            ),
            onClick: () => {}
          },
          {
            label: (
              <Box border="top" direction="row" pad="small">
                <Text className="spaced-text" size="small" weight="bold">
                  VIEW ALL NOTIFICATIONS
                </Text>
              </Box>
            )
          }
        ]}
      />
    </Box>
  )
}

export default NavbarNotifications
