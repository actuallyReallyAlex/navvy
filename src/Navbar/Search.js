import React, { Fragment, useState } from 'react'
import { Box, TextInput } from 'grommet'
import { Search } from 'grommet-icons'

const NavbarSearch = () => {
  const [searchValue, setSearchValue] = useState('')
  
  return (
    <Fragment>
      <Box margin={{ horizontal: 'small' }}>
        <Search />
      </Box>
      <Box align="center" direction="row" gap="small" width="50%">
        <TextInput
          margin
          onChange={e => setSearchValue(e.target.value)}
          placeholder="Search..."
          plain
          size="small"
          value={searchValue}
        />
      </Box>
    </Fragment>
  )
}

export default NavbarSearch
