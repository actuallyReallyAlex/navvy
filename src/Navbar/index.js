import React from 'react'
import { Box } from 'grommet'
import Search from './Search'
import Notifications from './Notifications'
import UserMenu from './UserMenu'

const Navbar = () => {
  return (
    <Box
      direction="row"
      fill="horizontal"
      height="70px"
      justify="between"
      style={{ boxShadow: '0 4px 2px -2px gray' }}
    >
      <Box align="center" direction="row">
      <Search />
      </Box>
      <Box
        align="center"
        direction="row"
        justify="end"
        pad={{ right: 'medium' }}
        width="33%"
      >
        <Notifications />
        <UserMenu />
      </Box>
    </Box>
  )
}

export default Navbar
