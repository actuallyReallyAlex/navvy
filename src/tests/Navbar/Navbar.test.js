import React from 'react'
import Navbar from '../../Navbar/index'
import { shallow } from 'enzyme'

const defaultProps = {}

describe('<Navbar />', () => {
  afterEach(() => {})

  test('Should render the <Navbar /> component.', () => {
    const component = shallow(<Navbar {...defaultProps} />)
    expect(component).toMatchSnapshot()
  })
})
