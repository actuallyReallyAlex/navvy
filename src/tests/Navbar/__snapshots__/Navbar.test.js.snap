// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> Should render the <Navbar /> component. 1`] = `ShallowWrapper {}`;
